webapp/src/containers/Rhs/projectDetails/index.tsx

```diff
@@ -87,6 +104,23 @@ const ProjectDetails = (projectDetails: ProjectDetails) => {
         };
     }, []);
 
+    useEffect(() => {
+        // Update subscription list only when it does not match with the current data
+        if (data !== subscriptionList) {
+            if (showAllSubscriptions) {
+                setSubscriptionList(data as SubscriptionDetails[]);
+            } else {
+                setSubscriptionList(getCurrentChannelSubscriptions(data as SubscriptionDetails[], currentChannelId));
+            }
+        }
+    }, [data]);
+
+    // Update subscription list on switching channels
+    useEffect(() => {
+        setShowAllSubscriptions(false);
```

Comment:
If the user is selected the option to show all the subscriptions, why are we resetting it? I guess, we shouldn't reset it unless the user manually toggles it.

Reply:
we are resetting the toggle when the user navigates to some other channel